### DISCUSSION TITLE
[solid-js] `experimental_prefetchInRender` with initial `enabled: false` makes page load indefinitely

### DISCUSSION THREAD

**Discussion by BierDav** [vikejs/vike#1990] [Discussion]
### Description

In our app we unfortunately have dependent queries and there are two ways we achieve this dependent behavior.
1. For simple usecases where the query params are not dependent on the query before an only the data is relevant eg. retrieving a JWT token: We use `(await dependent_query.promise)?.token` in the `queryFn` which works really well, but requires `experimental_prefetchInRender: true`
2. For advanced usecases, where the query params are dependent on the query before eg. retrieving list of users and than the detail of one of the users: We use `enabled: false` as long in the dependent query until the parent query has been resolved.

Here is a minimal reproduction: https://stackblitz.com/~/github.com/BierDav/vike-solid-query-bug?file=pages/index/Movies.tsx

Here you can clearly see that even though the query is enabled immedately after the query is initialized with `enabled: false` that and the query is even executed, but the page still remains loading.
https://github.com/user-attachments/assets/be57d4ad-111c-4bcb-89f9-ed52c6b51140

Unfortunately the second/dependent query is executed on the client as you can see wheny, but that likely due to this issue: https://github.com/TanStack/query/issues/7513

I hope someone can help me with that because this is real blocker for us switching to vike 🫶

**Comment by brillout** [comment]
Your question doesn't seem related to Vike. But feel free to elaborate if you do believe it has something to do with Vike.
